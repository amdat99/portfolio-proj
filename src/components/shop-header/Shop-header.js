import React, { useState } from "react";
import { Link } from "react-router-dom";

import SearchBox from "../search-box/Search-box";
import CartDropdown from "../cart/Cart-dropdown";
import CartButton from "../cart/Cart-button";

import { createStructuredSelector } from "reselect";
import { connect } from "react-redux";
import { selectCartHidden } from "../../redux/cart/cart.selectors";
import { fetchCategoryPending } from "../../redux/shop/shop.actions";
// import { selectCurrentUser } from '../../redux/user/user.selectors';

import "./Shop-header.scss";

export function ShopHeader({ hidden, fetchCategoryPending }) {
  const [shopDropdown, setShopDropdown] = useState(false);
  const [categories] = useState([
    "electronics",
    "clothes",
    "shoes",
    "food",
    "general",
  ]);

  const onDropdownToggle = () => {
    setShopDropdown(!shopDropdown);
  };
  return (
    <div className="shopheader-container">
      <div className="shopheader-links">
        <div
          className="shopheader-labels"
          id="shopheader-hover"
          onMouseLeave={onDropdownToggle}
          onMouseEnter={onDropdownToggle}
        >
          <Link to="/store"> SHOP</Link>
          {shopDropdown ? (
            <div className="shopheader-dropdown">
              {categories.map((category, i) => (
                <Link
                  to={`/category/${category.toLowerCase()}`}
                  className="shopheader-title"
                  onClick={() => fetchCategoryPending(category)}
                  key={i}
                >
                  {category}{" "}
                </Link>
              ))}
            </div>
          ) : null}
        </div>

        <div as="Link" className="shopheader-labels">
          <Link to="/listitems"> SEll ITEMS </Link>
        </div>
        {/* <div as= 'Link' className= 'shopheader-labels'><Link  to='/contact'> CONTACT US</Link></div> */}
        <CartButton />
        {hidden ? null : <CartDropdown />}
      </div>
      <Link to="/products">
        <SearchBox />
      </Link>
    </div>
  );
}

const mapStateToProps = createStructuredSelector({
  hidden: selectCartHidden,
});

const mapDispatchToProps = (dispatch) => ({
  fetchCategoryPending: (category) => dispatch(fetchCategoryPending(category)),
});
export default connect(mapStateToProps, mapDispatchToProps)(ShopHeader);
