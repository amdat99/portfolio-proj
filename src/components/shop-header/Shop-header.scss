.shopheader-container{
    height: 60px;
    width: 80%;
    display: flex;
    justify-content: space-between;
    margin-bottom: 50px;
    position: fixed;
    top: 0;
    left: 120px;
    z-index: 3;
    background: white;
    padding-top: 10px;
  
    @media screen  and (max-width: 800px ) {
        
        height: 60px;
        max-width: 500px;
        left: 45px;
        width: 90%;  
       
      
      }

.shopheader-dropdown{
    font-size: 14px;
    height: 100px;
    display: flex;
    flex-direction: column;
    padding: 7px;
    border: 1px solid black;
    background: white;
     position: fixed;
   z-index: 5;

   
   @media screen  and (max-width: 800px ) {
       
    left: 10px;
   padding:20px;
  }
   


}
    
.shopheader-profile{
    font-size: 14px;
    height: 100px;
    display: flex;
    flex-direction: column;
    padding: 5px;
    border: 1px solid black;
    background-color: white;
    position: fixed;
   
    z-index: 4
}
.shopheader-links{
    width: 50%;
    height: 100%;
    display: flex;
    align-items: center;
     
    }
.shopheader-labels{
   
    padding: 10px 15px;
    cursor: pointer;
    &:hover{
        transform: scale(1.1);
        cursor: pointer;
        opacity: 0.8;
    }
}}

.shopheader-label{
   
    padding: 10px 15px;
    cursor: pointer;
    &:hover{
        transform: scale(1.1);
        cursor: pointer;
        opacity: 0.8;
    }
}

.shopheader-title{
    &:hover{
        transform: scale(1.1);
        cursor: pointer;
        opacity: 0.8;
    }
}



