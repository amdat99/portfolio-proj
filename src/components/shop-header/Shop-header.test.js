import React from 'react';

import { shallow} from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import toJson from 'enzyme-to-json';

import {ShopHeader} from './Shop-header'

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
// describe('cart item', () => {
      

        let wrapper
        let wrapper2
        const mockCategories = ['shoes'];
        let mockfetchCategoryPending = jest.fn();
        let mockonDropdownToggle = jest.fn();
   

        
  beforeEach(() => {




   const mockProps = {
          categories: mockCategories,

            fetchCategoryPending: mockfetchCategoryPending,
            onDropdownToggle: mockonDropdownToggle
           
          }; 
          
           wrapper = toJson(shallow(<ShopHeader  {...mockProps} />))
           wrapper2 = shallow(<ShopHeader {...mockProps} />);
        })
        



it('expect to render cartDropdowncomponent', () => {
    expect(wrapper).toMatchSnapshot();
})


// it('should call category  when button clicks', () => {
//   wrapper2.find('.shopheader-title').simulate('click');
//   expect(mockfetchCategoryPending).toHaveBeenCalled();
// });

// it('should call togglecart  when button clicks', () => {
// wrapper2.find('[id="shopheader-hover"]').simulate('mouseenter');
//     expect(mockonDropdownToggle).toHaveBeenCalled();
//   });
  




