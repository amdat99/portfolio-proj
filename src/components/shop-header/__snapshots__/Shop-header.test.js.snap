// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render cartDropdowncomponent 1`] = `
<div
  className="shopheader-container"
>
  <div
    className="shopheader-links"
  >
    <div
      className="shopheader-labels"
      id="shopheader-hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <Link
        to="/store"
      >
         SHOP
      </Link>
    </div>
    <div
      as="Link"
      className="shopheader-labels"
    >
      <Link
        to="/listitems"
      >
         SEll ITEMS 
      </Link>
    </div>
    <div
      as="Link"
      className="shopheader-labels"
    >
      <Link
        to="/contact"
      >
         CONTACT US
      </Link>
    </div>
    <Connect(CartButton) />
    <Connect(CartDropdown) />
  </div>
  <Link
    to="/products"
  >
    <Connect(SearchBox) />
  </Link>
</div>
`;
